lab-jaren/test/game-router-test.js

```diff
@@ -0,0 +1,150 @@
+        expect(res.body.platforms).to.equal('Windows, Linux, Mac, PS4, Xbox One');
+        expect(res.body.ratingESRB).to.equal('Teen');
+        expect(res.body.releaseDate).to.equal('Feb 05, 2016');
+        expect(Boolean(res.body.id)).to.equal(true);
+        this.tempGameId = res.body.id;
+        done();
+      })
+      .catch(done);
+    });
+    it('invalid input should return 400 status code', done => {
+      superagent.post(`${baseURL}/api/games`)
+      .send({title: 'I am bread'})
+      .then(done)
+      .catch(err => {
+        expect(err.status).to.equal(400);
+        done();
+      })
+      .catch(done);
+    });
+  });
+  describe('testing GET /api/games/:id', function() {
+    before(done => {
+      superagent.post(`${baseURL}/api/games`)
+      .send(mockData)
+      .then(res => {
+        this.tempGameId = res.body.id;
+        done();
+      })
+      .catch(done);
+    });
+    after(done => {
+      Game.deleteById(this.tempGameId)
+      .then(() => done())
+      .catch(done);
+    });
+    it('valid id should return a game with status 200', done => {
+      superagent.get(`${baseURL}/api/games/${this.tempGameId}`)
+      .then(res => {
+        expect(res.status).to.equal(200);
+        expect(res.body.title).to.equal('XCOM 2');
+        expect(res.body.genre).to.equal('strategy/tactics');
+        expect(res.body.developer).to.equal('2K Games, Inc.');
+        expect(res.body.publisher).to.equal('Firaxis Games, Inc.');
+        expect(res.body.platforms).to.equal('Windows, Linux, Mac, PS4, Xbox One');
+        expect(res.body.ratingESRB).to.equal('Teen');
+        expect(res.body.releaseDate).to.equal('Feb 05, 2016');
+        expect(Boolean(res.body.id)).to.equal(true);
+        done();
+      })
+      .catch(done);
+    });
+    it('invalid id should return status 404', done => {
+      superagent.get(`${baseURL}/api/games/42`)
+      .then(done)
+      .catch(err => {
+        expect(err.status).to.equal(404);
+        done();
+      })
+      .catch(done);
+    });
+  });
+  describe('testing GET /api/games', function() {
+    before(done => {
+      superagent.post(`${baseURL}/api/games`)
+      .send(mockData)
+      .then(res => {
+        this.tempGameId = res.body.id;
+        done();
+      })
+      .catch(done);
+    });
+    after(done => {
+      Game.deleteById(this.tempGameId)
+      .then(() => done())
+      .catch(done);
+    });
+    it('should return an array containing .gitignore plus the IDs and status 200', done => {
+      superagent.get(`${baseURL}/api/games`)
+      .then(res => {
+        expect(res.status).to.equal(200);
+        expect(res.body).to.be.instanceof(Array);
+        expect(res.body.indexOf(this.tempGameId)).to.not.equal(-1);
+        done();
+      })
+      .catch(done);
+    });
+  });
+  describe('testing DELETE /api/games/:id', function() {
+    before(done => {
+      superagent.post(`${baseURL}/api/games`)
```

Comment:
My bad for giving not-the-best advice in the past about this before we started using persisting file storage. Duncan mentioned in lecture today that it's best to just utilize your ``createItem`` methods in the before block rather than going through the trouble of a ``POST`` request to the server. It saves more time and doesn't rely on connectivity to create the mock data.